### PR TITLE
tests: benchmarks: current_consumption: add qdec test

### PR DESCRIPTION
**Pull request by nordic-pikr**
Add add qdec current consumption test

### CODE REVIEW

File: tests/benchmarks/power_consumption/qdec/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+tests:
+  benchmarks.power_consumption.qdec_nrf54l:
+    platform_allow:
+      - nrf54l15pdk/nrf54l15/cpuapp
+    integration_platforms:
+      - nrf54l15pdk/nrf54l15/cpuapp
+    harness_config:
+      fixture: gpio_loopback
+      pytest_root:
+        - "${CUSTOM_ROOT_TEST_DIR}/test_measure_power_consumption.py::test_thread_suspend_resume_gpio_54L"
```
Comment (nordic-piks) [inline tests/benchmarks/power_consumption/qdec/testcase.yaml:14]: gpio?

File: tests/benchmarks/power_consumption/qdec/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  benchmarks.power_consumption.qdec_nrf54h:
+    sysbuild: true
+    platform_allow:
+      - nrf54h20dk/nrf54h20/cpuapp
+    integration_platforms:
+      - nrf54h20dk/nrf54h20/cpuapp
+    harness_config:
+      fixture: gpio_loopback
+      pytest_root:
+        - "${CUSTOM_ROOT_TEST_DIR}/test_measure_power_consumption.py::test_thread_suspend_resume_gpio_54H"
```
Comment (nordic-piks) [inline tests/benchmarks/power_consumption/qdec/testcase.yaml:24]: gpio?

File: tests/benchmarks/power_consumption/qdec/testcase.yaml (hunk 1 above)
Comment (nordic-pikr) [reply tests/benchmarks/power_consumption/qdec/testcase.yaml:14]: The limits set in this function are universal, there is no point in creating a new function that differs only in name

File: tests/benchmarks/power_consumption/qdec/testcase.yaml (hunk 2 above)
Comment (nordic-pikr) [reply tests/benchmarks/power_consumption/qdec/testcase.yaml:24]: as above

**Review by nordic-piks — APPROVED** [review]

**Comment by rlubos** [comment]
@nrfconnect/ncs-co-build-system Please review

File: tests/benchmarks/power_consumption/qdec/sysbuild.cmake [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Add radio core project
+ExternalZephyrProject_Add(
```
Comment (nordicjm) [inline tests/benchmarks/power_consumption/qdec/sysbuild.cmake:8]: do this properly as per https://github.com/nrfconnect/sdk-nrf/blob/main/sysbuild/Kconfig.netcore#L15 and https://github.com/nrfconnect/sdk-nrf/blob/main/sysbuild/netcore.cmake#L10

File: tests/benchmarks/power_consumption/qdec/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(qdec)
+
```
Comment (nordicjm) [inline tests/benchmarks/power_consumption/qdec/CMakeLists.txt:10]: if sysbuild is needed then follow https://github.com/nrfconnect/sdk-nrf/blob/main/tests/benchmarks/multicore/idle_with_pwm/CMakeLists.txt

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by nordic-piks — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**